### PR TITLE
fix: camel cased relationships do not resolve correctly

### LINKED ISSUE #297: Camel cased relationships do not resolve correctly
Currently the following code fails because the model cannot be resolved due to it's internal name being `style-sheet`.

```javascript
import { Model } from 'lux-framework';

class Website extends Model {
  static hasMany = {
    styleSheets: {
      inverse: 'website'
    }
  };
}

export default Website;
```

### PR DESCRIPTION
**Pull request by zacharygolba**
Closes #297

### CODE REVIEW

File: src/packages/database/index.js [hunk 1]
```diff
@@ -43,7 +45,7 @@ class Database {
   }
 
   modelFor(type: string): Class<Model> {
-    const model = this.models.get(type);
+    const model = this.models.get(normalizeModelName(type));
```
Comment (zacharygolba) [inline src/packages/database/index.js]: Before attempting to resolve a model the string is first normalized to ensure it is formatted the way model's are stored in `Database#models`.

File: src/packages/database/model/initialize-class.js [hunk 2]
```diff
@@ -269,7 +269,7 @@ export default async function initializeClass<T: Class<Model>>({
 
       Object.defineProperties(relationship, {
         model: {
-          value: store.modelFor(relatedModel || singularize(relatedName)),
+          value: store.modelFor(relatedModel || relatedName),
```
Comment (zacharygolba) [inline src/packages/database/model/initialize-class.js]: No need to `singularize` before calls to `store.modelFor` as normalization now takes place before the key is retrieved from the `Map` stored in `Database#models`.

File: src/packages/serializer/index.js [hunk 3]
```diff
@@ -386,7 +389,7 @@ class Serializer {
         [...this.hasOne, ...this.hasMany].reduce((hash, name) => ({
           ...hash,
 
-          [name]: (async () => {
+          [dasherize(underscore(name))]: (async () => {
```
Comment (zacharygolba) [inline src/packages/serializer/index.js]: `camelCase` relationship names now must be normalized before their added as keys to a [JSONAPI Relationships Object](http://jsonapi.org/format/#document-resource-object-relationships).

**Comment by adampash** [comment]
Looks good to me! Seems like you've cleaned up a lot of repetitive singularize-ing.

**Pull request merged**